File: framework/src/main/java/com/thornbirds/frameworkext/component/route/IRouter.java
package com.thornbirds.frameworkext.component.route;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;

/**
 * @author YangLi yanglijd@gmail.com
 */
public interface IRouter {
    boolean startPage(@NonNull String route);

    boolean startPage(@NonNull String route, @Nullable IPageParams params);

    boolean popPage();

    boolean popPage(@NonNull RouteComponentController page);
}

File: framework/src/main/java/com/thornbirds/frameworkext/component/route/IRouter.kt
package com.thornbirds.frameworkext.component.route

/**
 * @author YangLi yanglijd@gmail.com
 */
interface IRouter<T> {
    fun startPage(route: String, params: IPageParams? = null): Boolean
    fun popPage(page: T? = null): Boolean
}